src/auth/guards/jwt-auth.guard.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  ExecutionContext,
+  ForbiddenException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { AuthGuard } from '@nestjs/passport';
+import { Observable } from 'rxjs';
+import { IS_PUBLIC_KEY } from 'src/common/decorators/public.decorator';
+
+@Injectable()
+export class JwtAuthGuard extends AuthGuard('jwt') {
+  constructor(private reflector: Reflector) {
+    super();
+  }
+
+  public canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
```

Comment:
I think this method can return 3 types of value but you should choose just one of them

Reply:
thx
updated it in the next task
https://github.com/anticher/nodejs-st-22-rest-api/commit/e6a15f192de5cfde87e140e945d62b878f87da00